src/eom-application.c

```diff
@@ -343,13 +344,18 @@ eom_application_show_window (EomWindow *window, gpointer user_data)
 	if (!gtk_widget_get_realized (GTK_WIDGET (window)))
 		gtk_widget_realize (GTK_WIDGET (window));
 
-	if (timestamp <= 0)
-		timestamp = gdk_x11_get_server_time (gtk_widget_get_window (GTK_WIDGET (window)));
+	display = gdk_display_get_default ();
+#ifdef GDK_WINDOWING_X11
```

Comment:
I think you can reduce the scope of display, and put here `display`, inside `GDK_WINDOWING_X11`:

```
GdkDisplay *display;
display = gdk_display_get_default ();
```

...or you can use it directly, with `if (GDK_IS_X11_DISPLAY (gdk_display_get_default ()))`

Reply:
done